packages/core/src/components/List/List.tsx

```diff
@@ -0,0 +1,404 @@
+  const renderLeftIcon = (item) => {
+    return isValidElement(item.icon)
+      ? item.icon
+      : item.icon?.({
+          isSelected: item.selected,
+          isDisabled: item.disabled,
+        });
+  };
+
+  const renderSelectAll = () => {
+    const { selectAll, selectionConjunction } = labels;
+
+    const anySelected = !!selection?.length;
+    const allSelected = selection.length === list.length;
+
+    const selectionLabel = (
+      <HvTypography as="span">
+        {!anySelected ? (
+          <>
+            <b>{selectAll}</b>
+            {` (${list.length})`}
+          </>
+        ) : (
+          <>
+            <b>{selection.length}</b>
+            {`\xa0${selectionConjunction}\xa0`}
+            {list.length}
+          </>
+        )}
+      </HvTypography>
+    );
+
+    return (
+      <StyledSelectAllCheckBox
+        id={setId(id, "select-all")}
+        label={selectionLabel}
+        onChange={handleSelectAll}
+        className={clsx(
+          listClasses.selectAllSelector,
+          classes?.selectAllSelector
+        )}
+        indeterminate={anySelected && !allSelected}
+        checked={allSelected}
+      />
+    );
+  };
+
+  const renderItemText = (item) => {
+    const ItemText = wrapperTooltip(hasTooltips, item.label, item.label);
+
+    return !multiSelect && item.path ? (
+      <StyledLink
+        key={item.label}
+        route={item.path}
+        classes={{ a: clsx(listClasses.link, classes?.link) }}
+      >
+        <ItemText />
+      </StyledLink>
+    ) : (
+      <ItemText />
+    );
+  };
+
+  const renderMultiSelectItem = (item, itemId) => {
+    if (useSelector) {
+      const Selection = wrapperTooltip(
+        hasTooltips,
+        <StyledMultiSelectCheckBox
+          id={setId(itemId, "selector")}
+          label={item.label}
+          checked={item.selected}
+          disabled={item.disabled}
+          onChange={(evt) => handleSelect(evt, item)}
+          classes={{
+            root: clsx(listClasses.selectorRoot, classes?.selectorRoot),
+            container: clsx(
+              listClasses.selectorContainer,
+              classes?.selectorContainer
+            ),
+            label: clsx(listClasses.truncate, classes?.truncate),
+          }}
+        />,
+        item.label
+      );
+      return <Selection />;
+    }
+
+    return renderItemText(item);
+  };
+
+  const renderSingleSelectItem = (item, itemId) => {
+    if (useSelector) {
+      const Selection = wrapperTooltip(
+        hasTooltips,
+        <StyledSingleSelectRadio
+          id={setId(itemId, "selector")}
+          label={item.label}
+          checked={item.selected}
+          disabled={item.disabled}
+          classes={{
+            root: clsx(listClasses.selectorRoot, classes?.selectorRoot),
+            container: clsx(
+              listClasses.selectorContainer,
+              classes?.selectorContainer
+            ),
+            label: clsx(listClasses.truncate, classes?.truncate),
```

Comment:
The [CSS from the truncate class](https://github.com/lumada-design/hv-uikit-react/blob/10621b6afb337e36ea171aae7c735beda015de30/packages/core/src/List/styles.js#L28) is missing in the component `StyledSingleSelectRadio`.